### PR TITLE
Fixing cli bugs for --brocfile param

### PR DESCRIPTION
**Pull request by chrmod**
So my previous PR was not so good in the end. Here goes fixes and some tests to cover them.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -23,19 +23,19 @@ module.exports = function broccoliCLI(args) {
     .description('start a broccoli server')
     .option('--port <port>', 'the port to bind to [4200]', 4200)
     .option('--host <host>', 'the host to bind to [localhost]', 'localhost')
-    .option('--brocfile-path <path>', 'the path to brocfile')
+    .option('--brocfile <path>', 'the path to brocfile')
```
Comment (rwjblue) [inline lib/cli.js]: We decided to change this to `--brocfile-path`, can you bring that back?

Comment (chrmod) [reply lib/cli.js]: done. can you please update https://github.com/broccolijs/broccoli/issues/331 to reflect your decisions. 
I plan to implement other params now, so would like to be sync :-)

Comment (rwjblue) [reply lib/cli.js]: No problem!  I updated it (we only discussed `--brocfile-path` though)...

File: lib/cli.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ module.exports = function broccoliCLI(args) {
     .action(options => {
       actionPerformed = true;
 
-      const builder = getBuilder(options.brocfilePath);
+      const builder = getBuilder({ brocfilePath: options.brocfilePath });
```
Comment (rwjblue) [inline lib/cli.js]: I'm unsure why we need to make a second object just to pass the single option into it. Any reason not to do either `getBuilder(options)` or `getBuilder(options.brocfilePath)` here?

Comment (chrmod) [reply lib/cli.js]: was planning to pass more options here, like `cwd`. But have to see how other options will be passed around to builder/watcher. As `build` and `serve` are going to share options, probably this is going to change.

Comment (chrmod) [reply lib/cli.js]: if you prefer I can simplify it now and change in upcoming PRs

Comment (rwjblue) [reply lib/cli.js]: Nah, seems totally fine, I was mostly just curious.  Answer is: planning more changes that will leverage the object. 👍

**Pull request merged**